bin/check-titles

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+from __future__ import annotations
+
+import difflib
+import os.path
+
+
+def main() -> int:
+    with open('README.md') as readme:
+        for line in readme:
+            if line.startswith('0. '):
+                break
+
+        for idx, line in enumerate(readme, 1):
+            title = line[3:]
+
+            ep_file = f'sample_code/ep{idx:03}/README.md'
+            if not os.path.isfile(ep_file):
+                print(f'Invalid filename: {ep_file}')
+                continue
+
+            with open(ep_file) as ep_readme:
+                for line in ep_readme:
+                    ep_title = line[2:]
+                    if title == ep_title:
+                        break
+
+                    print(f'\n{ep_file}\n{title}\n')
+                    differ = difflib.Differ()
+                    diff = differ.compare(title.split(), ep_title.split())
+                    for el in diff:
+                        if el.startswith(('+', '-')):
+                            print(el[2:])
```

Comment:
True. At some point I tried to produce git-like diffs with colors and bolding for the difference.

Regarding CI - `main` will have to return non-zero at some point in order to be useful there.
I wrote the script just as a convenience tool for me to check the current inconsistencies,
It's included just in case if we ever need it again in the future I guess.